assets/src/edit-story/components/canvas/carrousel/index.js

```diff
@@ -30,13 +31,47 @@ const Page = styled.a`
 `;
 
 function Canvas() {
-	const { state: { pages, currentPageIndex }, actions: { setCurrentPage } } = useStory();
-	const handleClickPage = ( page ) => () => setCurrentPage( { pageId: page.id } );
+	const { state: { pages, currentPageIndex }, actions: { setCurrentPage, arrangePage } } = useStory();
+	const getArrangeIndex = ( sourceIndex, dstIndex, position ) => {
+		// If the dropped element is before the dropzone index then we have to deduct
+		// that from the index to make up for the "lost" element in the row.
+		const indexAdjustment = sourceIndex < dstIndex ? -1 : 0;
+		if ( 'left' === position.x ) {
+			return dstIndex + indexAdjustment;
+		}
+		return dstIndex + 1 + indexAdjustment;
+	};
+	const handleClickPage = ( page ) => setCurrentPage( { pageId: page.id } );
 	return (
 		<List>
-			{ pages.map( ( page, index ) => (
-				<Page key={ index } onClick={ handleClickPage( page ) } isActive={ index === currentPageIndex } />
-			) ) }
+			{ pages.map( ( page, index ) => {
+				const onDrop = ( evt, position ) => {
```

Comment:
Why is the function defined in the map. Wouldn't that mean it was redefined on every loop. Can't this function be moved?

Reply:
Moved this by adding `pageIndex` attribute to dropzone. Might be useful elsewhere anyway for determining which page a dropzone belongs to.